### ISSUE TITLE
E201 / E202 false positives inside f-strings for Python 3.12

### ISSUE THREAD

**Issue by vanschelven** [PyCQA/pycodestyle#1260]
I have f-strings like so:

`f"{ parts.scheme }://{ public_key }@{ parts.hostname }"`

On Python 3.11 and below pycodestyle (via flake8) is happy with them. On Python 3.12 I get E201 and E202 (`whitespace after '{'`, `whitespace before '}'`)

This error is surprising to me, and "fixing" it makes the code much harder to read. Thoughts:

* Pep8 cannot really be a guidance here, because it predates f-strings (by far)
* Is it the position of the `pycodestyle` developers that whitespace-rules should be applied equally inside and outside f-strings? 
* Whatever the position on what the correct behavior is, it's "surprising" (i.e. a bug) that the behavior depends on your Python version.
* On a [more or less recent discussion about the modification of the f-string syntax, a Python core dev explicitly mentions the OK-ness of whitespace inside f-strings](https://github.com/python/cpython/issues/80998#issuecomment-1093822745)

> Finally, and this is the best part: what if you want whitespace around the equals sign? [..] no problem; whitespace is preserved from the original text inside the curly braces [..]  we also explicitly permit, and preserve, whitespace.

Similar discussions (sometimes prematurely marked as resolved):

* https://stackoverflow.com/questions/77224809/does-f-string-float-formatting-really-violate-pep-8
* https://github.com/PyCQA/flake8/issues/1876
* https://github.com/PyCQA/pycodestyle/issues/1201

====

```
>>> pycodestyle.__version__
'2.12.1'
```

**Comment by asottile** [comment]
you found the duplicates with answers please don't make new duplicates
